test/runtests.jl

```diff
@@ -1,8 +1,22 @@
 using ShiftedArrays, Test
 using AbstractFFTs 
+use_cuda = false;  # set this to true to test ShiftedArrays for the CuArray datatype
+if (use_cuda)
+    using CUDA
+    CUDA.allowscalar(true); # needed for some of the comparisons
+end
```

Comment:
Maybe we could use https://github.com/JuliaGPU/GPUArrays.jl/blob/4278412a6b9b1d859c290232a9f8223eb4416d1e/lib/JLArrays/src/JLArrays.jl#L6 to test without a GPU? Also, maybe we can try and use `@allowscalar` to selectively allow scalar indexing where it's needed.

Reply:
I did not want CUDA to be a dependency on testing, if we move `@allowscalar` outside the if clause, we would need to always have CUDA loaded for testing. If `JLArrays.jl` is a lightweight way of doing this, then this may be a good idea.